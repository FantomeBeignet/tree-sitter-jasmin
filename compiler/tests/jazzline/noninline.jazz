fn f(reg u64 x) -> reg u64 {
   x += 1;
   return x;
}

fn g(reg u64 x) -> reg u64 {
   x = f(x);
   x = f(x);
   return x;
}