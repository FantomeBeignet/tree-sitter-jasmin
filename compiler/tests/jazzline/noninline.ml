open Common

let () =
  parse_and_print "noninline.jazz"  "f";
  parse_and_print "noninline.jazz"  "g"
