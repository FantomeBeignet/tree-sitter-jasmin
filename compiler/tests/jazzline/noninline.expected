proc f(x_141@uint64, x_137@uint64) = 
{
 true && true
}
adds TMP_____152@uint1 x_141@uint64 (1)@uint64 x_137@uint64;
{
 true && true
}
 proc g(x_179@uint64, x_180@uint64, x_170@uint64) = 
{
 true && true
}
assert true && true;
assume true && true;
assert true && true;
assume true && true;
{
 true && true
}
 